census/estimator/trainer/model.py

```diff
@@ -215,29 +214,20 @@ def example_serving_input_fn():
       example_bytestring,
       tf.feature_column.make_parse_example_spec(INPUT_COLUMNS)
   )
-  features = {
-      key: tf.expand_dims(tensor, -1)
-      for key, tensor in feature_scalars.iteritems()
-  }
-  return tf.contrib.learn.InputFnOps(
+  return tf.estimator.export.ServingInputReceiver(
       features,
-      None,  # labels
       {'example_proto': example_bytestring}
   )
 
-
+# [START serving-function]
 def json_serving_input_fn():
   """Build the serving inputs."""
   inputs = {}
   for feat in INPUT_COLUMNS:
     inputs[feat.name] = tf.placeholder(shape=[None], dtype=feat.dtype)
 
-  features = {
-      key: tf.expand_dims(tensor, -1)
-      for key, tensor in inputs.iteritems()
-  }
-  return tf.contrib.learn.InputFnOps(features, None, inputs)
-
+  return tf.estimator.export.ServingInputReceiver(inputs, inputs)
```

Comment:
tf.estimator.export.build_raw_serving_input_receiver_fn

Reply:
Interesting this exists :) There is actually a copy of feature_dict happening in the code which is being done defensively. But there is a TODO to get rid of it. I am bit hesitant to add another layer of indirection which is not doing much. Infact I am not sure why the API exists.